napari/layers/shapes/_tests/test_shapes_mouse_bindings.py

```diff
@@ -1,4 +1,5 @@
-import collections
+from dataclasses import dataclass, field
+from typing import List, Optional, Tuple, Union
```

Comment:
@brisvag sorry to be a pain, but I feel like I want to pull out these changes, which as far as I can tell have nothing to do with this PR (you don't seem to use them to make a specific test for this PR's issue? Just improve the readability of existing tests? Or did I miss the one new test? Which only highlights the problem tbh 😅). It's annoying, but I think having small, self-contained PRs is critical later when chasing up some bug or design decision...

Then, ideally, you would use the new infra to test this PR's fixes, but I don't really know how challenging that is. Seems significant. 😅

Reply:
I would in fact prefer... 🙏 I've done a lot of git archeology in my time and big unrelated changes are always annoying to come across.

Reply:
Ok, reverted except for the 1 needed change and another where the test was essentially asserting the wrong thing.

Reply:
Pulled out the changes to #5917!